src/dmd/e2ir.d

```diff
@@ -5750,6 +5773,8 @@ private elem *toElemStructLit(StructLiteralExp sle, IRState *irs, TOK op, Symbol
             //e = el_combine(e, ev);
             elem_setLoc(e, sle.loc);
         }
+        if (forcetype)
+            return Lreinterpret(sle.loc, e, forcetype);
```

Comment:
This is not covered by tests.

Reply:
It is, otherwise the runnable_cxx tests wouldn't pass.